metrics/report/report_dockerfile/footprint-density.R

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env Rscript
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+library(ggplot2)	# ability to plot nicely
+library(gridExtra)	# So we can plot multiple graphs together
+suppressMessages(suppressWarnings(library(ggpubr)))		# for ggtexttable
+suppressMessages(library(jsonlite))		# to load the data
+
+testname="Container PSS footprint"
+
+resultsfiles=c(
+	paste("footprint-busybox", test_name_extra, ".json", sep=""),
+	paste("footprint-mysql", test_name_extra, ".json", sep=""),
+	paste("footprint-elasticsearch", test_name_extra, ".json", sep="")
+)
+
+data=c()
+stats=c()
+rstats=c()
+rstats_names=c()
+
+# For each set of results
+for (currentdir in resultdirs) {
+	count=1
+	dirstats=c()
+	for (resultsfile in resultsfiles) {
+		fname=paste(inputdir, currentdir, resultsfile, sep="")
+		if ( !file.exists(fname)) {
+			warning(paste("Skipping non-existent file: ", fname))
+			next
+		}
+
+		# Derive the name from the test result dirname
+		datasetname=basename(currentdir)
+
+		# Import the data
+		fdata=fromJSON(fname)
+
+		testname=paste(datasetname, fdata$Config$payload, sep="-")
+
+		cdata=data.frame(avail_mb=as.numeric(fdata$Results$system$avail)/(1024*1024))
+		cdata=cbind(cdata, avail_decr=as.numeric(fdata$Results$system$avail_decr))
+		cdata=cbind(cdata, count=seq_len(length(cdata[, "avail_mb"])))
+		cdata=cbind(cdata, testname=rep(testname, length(cdata[, "avail_mb"]) ))
+
+		# Gather our statistics
+		sdata=data.frame(num_containers=length(cdata[, "avail_mb"]))
+		# Pick out the last avail_decr value - which in theory should be
+		# the most we have consumed...
+		sdata=cbind(sdata, mem_consumed=cdata[, "avail_decr"][length(cdata[, "avail_decr"])])
+		sdata=cbind(sdata, avg_bytes_per_c=sdata$mem_consumed / sdata$num_containers)
+		sdata=cbind(sdata, runtime=testname)
+
+		# Store away as a single set
+		data=rbind(data, cdata)
+		stats=rbind(stats, sdata)
+
+		s = c(
+			"Test"=testname,
+			"n"=sdata$num_containers,
+			"size"=(sdata$mem_consumed) / 1024,
+			"kb/n"=round((sdata$mem_consumed / sdata$num_containers) / 1024, digits=1),
+			"n/Gb"= round((1*1024*1024*1024) / (sdata$mem_consumed / sdata$num_containers), digits=1)
+		)
+
+		rstats=rbind(rstats, s)
+		# Store away some stats for the text table
+#		dirstats[count]=round(fdata.mean, digits=2)
+
+		count = count + 1
+	}
+#	rstats=rbind(rstats, dirstats)
+#	rstats_names=rbind(rstats_names, datasetname)
+}
+
+#rstats=cbind(rstats_names, rstats)
+
+#unts=c("Kb", "Kb")
+
+# If we have only 2 sets of results, then we can do some more
+# stats math for the text table
+if (length(resultdirs) == 2) {
+	# This is a touch hard wired - but we *know* we only have two
+	# datasets...
+#	diff=c("diff")
+#	val = ((as.double(rstats[1,2]) / as.double(rstats[2,2])) * 100) - 100
+#	diff[2] = round(val, digits=2)
+#	val = ((as.double(rstats[1,3]) / as.double(rstats[2,3])) * 100) - 100
+#	diff[3] = round(val, digits=2)
+#	rstats=rbind(rstats, diff)
```

Comment:
Much commented out code :)

Reply:
yep. I'd left it in the hope of adding back in the 'compare two sets of data' feature, but didn't get to it. I'll add that to the 'todo' issue #601